src/explorer/SiteTreeItem.ts

```diff
@@ -109,19 +109,30 @@ export abstract class SiteTreeItem implements IAzureParentTreeItem {
         const correlationId = getRandomHexString(10);
         telemetryProperties.correlationId = correlationId;
 
-        const workspaceConfig: WorkspaceConfiguration = workspace.getConfiguration(constants.extensionPrefix, Uri.file(fsPath));
+        const siteConfig: WebSiteModels.SiteConfigResource = await this.client.getSiteConfig();
+        let deploymentFilePath: string;
+        if (fsPath) {
+            deploymentFilePath = fsPath;
```

Comment:
nit: Not sure I see the point of deploymentFilePath. You could just do:

```
if (!fsPath) {
    if (siteConfig.linuxFxVersion && siteConfig.linuxFxVersion.toLowerCase().startsWith('tomcat')) {
        fsPath = ...
    } else {
        fsPath = ...
    }
}
```

Reply:
Find that if I remove the `deploymentFilePath` the method `appservice.deploy()` will have an error said:
`Argument of type 'string | undefined' is not assignable to parameter of type 'string'`

Other places are fine.

Maybe we need change the method signature in `appservice.deploy()` and throw an error when the fsPath is `undefined`?